spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/GoogleConfigPropertySourceLocator.java

```
@@ -0,0 +1,138 @@
+		Assert.notNull(projectIdProvider, "Project ID provider cannot be null");
+		Assert.notNull(defaultProperties, "Config client properties cannot be null");
+		this.credentials = credentialsProvider.getCredentials();
+		this.projectId = projectIdProvider.getProjectId();
+		this.defaultProperties = defaultProperties;
+	}
+
+	private RestTemplate getSecureRestTemplate() throws Exception {
+		SimpleClientHttpRequestFactory requestFactory = new SimpleClientHttpRequestFactory();
+		requestFactory.setReadTimeout((60 * 1000 * 3) + 5000);
+		RestTemplate template = new RestTemplate(requestFactory);
+		Map<String, String> headers = new HashMap<>();
+		Map<String, List<String>> requestMetadata = this.credentials.getRequestMetadata();
+		Assert.isTrue(requestMetadata.containsKey(AUTHORIZATION_HEADER), "Authorization header required");
+
+		List<String> authorizationHeaders = requestMetadata.get(AUTHORIZATION_HEADER);
+
+		String bearerToken = null;
+		for (String authorizationHeader : authorizationHeaders) {
+			if (authorizationHeader.startsWith(BEARER_TOKEN_PREFIX)) {
+				bearerToken = authorizationHeader;
+				break;
+			}
+		}
+		Assert.notNull(bearerToken, "Valid Bearer token not found");
+
+		headers.put(AUTHORIZATION_HEADER, bearerToken);
+
+		template.setInterceptors(Collections
+				.singletonList(new ConfigServicePropertySourceLocator.GenericRequestHeaderInterceptor(headers)));
+		return template;
+	}
+
+	private GoogleConfigEnvironment getRemoteEnvironment(RestTemplate restTemplate, ConfigClientProperties properties)
+			throws IOException, HttpClientErrorException {
+		String name = properties.getName();
+		String profile = properties.getProfile();
+
+		ResponseEntity<GoogleConfigEnvironment> response = restTemplate.exchange(
+				RUNTIMECONFIG_API_ROOT + ALL_VARIABLES_PATH, HttpMethod.GET,
+				null, GoogleConfigEnvironment.class, this.projectId, name, profile);
+
+		Assert.isTrue(response != null && response.getStatusCode().is2xxSuccessful(),
```

Should we throw an exception other than `IllegalArgumentException` here?

---

Agree, fixed.